pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -214,40 +234,239 @@ func (c *MCRouteController) Run(stopCh <-chan struct{}) {
-// function in order to read and process a message on the workqueue.
-func (c *MCRouteController) worker() {
-	for c.processNextWorkItem() {
+// gatewayWorker is a long-running function that will continually call the processNextGatewayItem
+// function in order to read and process a message on the gatewayQueue.
+func (c *MCDefaultRouteController) gatewayWorker() {
+	for c.processNextGatewayItem() {
+	}
+}
+
+// ciImportWorker is a long-running function that will continually call the processNextClusterInfoImportItem
+// function in order to read and process a message on the ciImportQueue.
+func (c *MCDefaultRouteController) ciImportWorker() {
+	for c.processNextClusterInfoImportItem() {
+	}
+}
+
+func (c *MCDefaultRouteController) processNextClusterInfoImportItem() bool {
+	obj, quit := c.ciImportQueue.Get()
+	if quit {
+		return false
+	}
+	defer c.ciImportQueue.Done(obj)
+
+	ciImp, ok := obj.(string)
+	if !ok {
+		c.ciImportQueue.Forget(obj)
+		klog.InfoS("Expected string in work queue but got", "object", obj)
+		return true
+	}
+	if c.wireGuardConfig != nil {
+		if err := c.updateWireGuardPeers(ciImp); err != nil {
+			c.ciImportQueue.AddRateLimited(ciImp)
+			klog.ErrorS(err, "Error updating WireGuard peers, requeuing", "key", ciImp)
+		}
+	}
+
+	if err := c.syncMCFlows(); err == nil {
+		c.ciImportQueue.Forget(obj)
+	} else {
+		// Put the item back on the workqueue to handle any transient errors.
+		c.ciImportQueue.AddRateLimited(ciImp)
+		klog.ErrorS(err, "Error syncing key, requeuing", "key", ciImp)
 	}
+	return true
 }
 
-func (c *MCRouteController) processNextWorkItem() bool {
-	obj, quit := c.queue.Get()
+func (c *MCDefaultRouteController) processNextGatewayItem() bool {
+	obj, quit := c.gatewayQueue.Get()
 	if quit {
 		return false
 	}
-	defer c.queue.Done(obj)
+	defer c.gatewayQueue.Done(obj)
 
-	if k, ok := obj.(string); !ok {
-		c.queue.Forget(obj)
+	gateway, ok := obj.(string)
+	if !ok {
+		c.gatewayQueue.Forget(obj)
 		klog.InfoS("Expected string in work queue but got", "object", obj)
 		return true
-	} else if err := c.syncMCFlows(); err == nil {
-		c.queue.Forget(k)
+	}
+
+	if c.wireGuardConfig != nil {
+		if err := c.reconcileWireGuard(); err != nil {
+			klog.ErrorS(err, "Error when reconcile WireGuard, requeuing", "key", gateway)
+			c.gatewayQueue.AddRateLimited(gateway)
+		}
+	}
+	if err := c.syncMCFlows(); err == nil {
+		c.gatewayQueue.Forget(obj)
 	} else {
 		// Put the item back on the workqueue to handle any transient errors.
-		c.queue.AddRateLimited(k)
-		klog.ErrorS(err, "Error syncing key, requeuing", "key", k)
+		c.gatewayQueue.AddRateLimited(gateway)
+		klog.ErrorS(err, "Error syncing key, requeuing", "key", gateway)
 	}
 	return true
 }
 
-func (c *MCRouteController) syncMCFlows() error {
+func (c *MCDefaultRouteController) reconcileWireGuard() error {
+	c.wireGuardMutex.Lock()
+	defer c.wireGuardMutex.Unlock()
+	gateways, err := c.gwLister.Gateways(c.namespace).List(labels.Everything())
```

Comment:
Lan added a func `getActiveGateway`. You should use the func to get the Gateway.

Reply:
Done